pennylane/tape/interfaces/autograd.py

```diff
@@ -160,6 +174,13 @@ def _execute(self, params, device):
             p.numpy() if isinstance(p, np.tensor) else p for p in self._all_parameter_values
         ]
 
+        if self.do_not_execute_device_as_this_is_the_needless_autograd_forward_pass:
+            self.do_not_execute_device_as_this_is_the_needless_autograd_forward_pass = False
+            # all that counts here is that we return somthing that has the right shape
+            # and that we return only after setting self._all_params_unwrapped as this
+            # is needed in the subsequent call to self.vjp() by autograd
+            return np.zeros(len(self.observables))

```

Comment:
I am not sure this is the best way of determining the shape but the first I could come up with that worked in my internal tests. Maybe @josh146 knows a better way to predict the output shape without running the device? As a last resort one could run one of the computations that are then anyway needed in `vjp()` here, determine the shape from that and then use that result (which is passed as `ans` to `vjp()`) to save an execution during `vjp()`.

Reply:
I see that some tests are failing because of this, so we need a better way to predict the output shape.